packages/taquito/src/contract/rpc-estimate-provider.ts

```diff
@@ -150,15 +174,22 @@ export class RPCEstimateProvider extends OperationEmitter implements EstimationP
    *
    * @param OriginationOperation Originate operation parameter
    */
-  async originate({ fee, storageLimit, gasLimit, ...rest }: OriginateParams) {
+  async originate({ fee, storageLimit, gasLimit, ...rest }: OriginateParams, estimationOptions: EstimationOptions = { includeRevealOperation: true }) {
     const pkh = await this.signer.publicKeyHash();
     const DEFAULT_PARAMS = await this.getAccountLimits(pkh);
     const op = await createOriginationOperation(
       await this.context.parser.prepareCodeOrigination({
-      ...rest,
-      ...mergeLimits({ fee, storageLimit, gasLimit }, DEFAULT_PARAMS),
-    }));
-    return (await this.createEstimate({ operation: op, source: pkh }))[0];
+        ...rest,
+        ...mergeLimits({ fee, storageLimit, gasLimit }, DEFAULT_PARAMS),
+      })
+    );
+    const isRevealNeeded = await this.isRevealOpNeeded([op], pkh);
+    const ops = isRevealNeeded ? await this.addRevealOp([op], pkh) : op;
```

Comment:
The isRevealOpNeeded method validates if the account (tz1..) is already revealed or not by making a call to the RPC. If the account is not revealed we need to include a reveal operation because we simulate the operation with another call to the RPC (here: const { opResponse } = await this.simulate(operation);) If the account is not revealed and we do not include a reveal operation, this call fail. So we need to include a reveal operation everytime if it is needed.